game-core/src/main/java/games/strategy/triplea/delegate/DiceRoll.java

```diff
@@ -219,37 +171,40 @@ public static DiceRoll rollDice(
       final String annotation,
       final CombatValue combatValueCalculator) {
 
+    final DiceRoll diceRoll;
+    final PowerStrengthAndRolls unitPowerAndRollsMap =
+        PowerStrengthAndRolls.build(units, combatValueCalculator);
     if (Properties.getLowLuck(bridge.getData().getProperties())) {
-      return rollDiceLowLuck(units, player, bridge, annotation, combatValueCalculator);
+      diceRoll = rollDiceLowLuck(unitPowerAndRollsMap, player, bridge, annotation);
+    } else {
+      diceRoll = rollDiceNormal(unitPowerAndRollsMap, player, bridge, annotation);
     }
-    return rollDiceNormal(units, player, bridge, annotation, combatValueCalculator);
+
+    bridge
+        .getHistoryWriter()
+        .addChildToEvent(annotation + " : " + MyFormatter.asDice(diceRoll), diceRoll);
+    return diceRoll;
   }
 
   /** Roll dice for units using low luck rules. */
   private static DiceRoll rollDiceLowLuck(
-      final Collection<Unit> unitsList,
+      final TotalPowerAndTotalRolls unitPowerAndRollsMap,
```

Comment:
`unitPowerAndRollsMap` is an unexpected name for this parameter as the type is not a map. Any objections to renaming it?